### PR TITLE
+slick #1329 add flatten to DBIOAction

### PR DESCRIPTION
**Pull request by hepin1989**

### CODE REVIEW

**Comment by hepin1989** [comment]
refs https://github.com/slick/slick/issues/1329

**Comment by cvogt** [comment]
cool stuff :). Could you amend the commit removing the unrelated whitespace changes?

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala [hunk 1]
```diff
@@ -18,18 +19,47 @@ class ActionTest extends AsyncTest[RelationalTestDB] {
...
+      } yield maybeMax match {
+          case Some(max) =>
+            ts.filter(_.a === max).result
+          case None =>
+            q1.result
+        }
+      f1 <- db.run(q2.flatten)
+      _ = f1 shouldBe Seq(5)
+    } yield ()
+  }
```
Comment (cvogt) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala]: looks a bit complicated, wouldn't a simpler test suffices, like `ts.result.map(_ => ts.result).flatten shouldBe List(1, 2, 3, 4, 5)` ?

Comment (hepin1989) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala]: In fact I did an Ctrl-C+V:)
for the format ,IDEA did that:(.

**Comment by cvogt** [comment]
@hepin1989 to get rid of the whitespace changes: if you do `git reset --mixed HEAD^^` you basically get the changes of both commits locally removing the commity. Then use `git add -p` to only add the non-whitespace ones. Then `git commit`. Then `git push -f` to update the PR.

**Comment by hepin1989** [comment]
@cvogt I have to change to IDEA to powersave mode :(.that's why I want the scalaform.thank you for you git command.

**Comment by cvogt** [comment]
LGTM. @szeiger ?

**Comment by szeiger** [comment]
@hepin1989 "Preferences -> Editor -> General -> Other -> Strip trailing spaces on Save: Modified Lines"

**Comment by cvogt** [comment]
cool stuff @hepin1989  :)

**Comment by hepin1989** [comment]
@cvogt @szeiger thanks for slick:)

**Pull request merged**